pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -53,6 +56,10 @@ class EarlyStopping(Callback):
             monitored has stopped decreasing and in ``'max'`` mode it will stop when the quantity
             monitored has stopped increasing.
         strict: whether to crash the training if `monitor` is not found in the validation metrics.
+        check_finite: Stops training when the monitor becomes NaN or infinite. Set this argument to ``False``
+            if this behavior is undesired.
+        stopping_threshold: Stop training immediately once the monitored quantity reaches this threshold.
+        divergence_threshold: Stop training as soon as the monitored quantity becomes worse than this threshold.
```

Comment:
we could use `stop_limit` and `stop_loss` to follow common financial terms

https://www.investopedia.com/articles/active-trading/091813/which-order-use-stoploss-or-stoplimit-orders.asp

Reply:
@jlperla  what do you think of this name suggestion?

Reply:
But might not be `loss`, and most people don't know finance. These are basically optimizer settings,which is more universal. 

I think sticking with optimizer style lingo is ideal. Divergence is safe and says what it means . Normally one would call the success criteria as tolerances for optimizers. But that is because they are always comparing something (eg a value itself, changes in that value, or first order conditions) to zero.

Since this could presumably compare stopping for this other than close to zero(especially if you are tracking something where a larger number is better) , I think threshold is probably more general. But open minded of course

Reply:
Could also go super simple and go with `min_threshold` and `max_threshold`

Reply:
No because that implies a direction to them.

Reply:
IMO, the name are good right now.

Reply:
this is not clear to me, it is you have some converging sequence so it stops when it starts diverse again? shall it be some patience for noise presence reason?
or natively it can be observing  training and validation measure and stop overfitting - when these twos tart diverse

Reply:
If something is diverging, it is because you are in some sort of local minima or outside of an attractor and it could only return with some massive jumps (i.e. emulation in simulating annealing you are way off in the boons for your optima... in theory it come come back, but it might takes months.  You are better off just restarting).  So patience isn't the right thing to think of for that.